containers/cpp/.devcontainer/Dockerfile

```diff
@@ -2,6 +2,20 @@
 ARG VARIANT=debian-11
 FROM mcr.microsoft.com/vscode/devcontainers/cpp:0-${VARIANT}
 
+# [Optional] Install CMake version different from base image.
+# [Choice] CMake Version: none, 3.21.5, 3.22.2, or versions from https://cmake.org/download/
+ARG REINSTALL_CMAKE_FROM_SOURCE="false"
+ARG CMAKE_VERSION="none"
+
+# Optionally install the cmake for vcpkg
+COPY library-scripts/reinstall-cmake.sh /usr/local/etc/vscode-dev-containers 
```

Comment:
nit: Not sure if `/usr/local/etc/vscode-dev-containers` is the preferred location for 'tmp' files. I've seen other containers just putting those under some other [dir](https://github.com/microsoft/vscode-dev-containers/blob/a31634f072eadf80b0b01db47332fe3d7126ed06/containers/azure-cli/.devcontainer/Dockerfile#L13)

Reply:
agreed, change made :)